build.gradle.kts

```diff
@@ -23,12 +23,22 @@ version = providers.gradleProperty("VERSION_NAME").get()
 group = providers.gradleProperty("GROUP").get()
 description = providers.gradleProperty("POM_DESCRIPTION").get()
 
+// Release artifacts target Java 11; however, some tests require Java 17+
+val testJdk = providers.gradleProperty("testJdk").getOrElse("17")
+
 dokka {
   dokkaPublications.html {
     outputDirectory = rootDir.resolve("docs/api")
   }
 }
 
+java {
+  toolchain {
+    // Use Java 21 for building the project (a recent LTS) to minimize the chances of bugs in the compiler
+    languageVersion = JavaLanguageVersion.of(21)
```

Comment:
Intended for not using toolchain, due to some reasons like https://jakewharton.com/gradle-toolchains-are-rarely-a-good-idea/.